app/services/sso/openid_connect.rb

```diff
@@ -0,0 +1,59 @@
+module SSO
+  class OpenidConnect < Base
+    delegate :session, :to => :controller
+    attr_reader :current_user
+
+    def available?
+      controller.api_request? && bearer_token_set? && valid_issuer?
+    end
+
+    def authenticate!
+      payload = jwt_token.decode
+      return nil if payload.nil?
+      user = find_or_create_user_from_jwt(payload)
+      @current_user = user
+      update_session(payload)
+      user&.login
+    end
+
+    def authenticated?
+      self.user = User.current.presence || authenticate!
+    end
+
+    private
+
+    def jwt_token
+      @jwt_token ||= jwt_token_from_request
+    end
+
+    def jwt_token_from_request
+      token = request.authorization.split(' ')[1]
+      OidcJwt.new(token)
+    end
+
+    def bearer_token_set?
+      request.authorization.present? && request.authorization.start_with?('Bearer')
+    end
+
+    def valid_issuer?
+      payload = jwt_token.decoded_payload
+      return false unless (payload.key?('iss') && payload['iss'] == Setting['oidc_issuer'])
+      true
```

Comment:
can't this just be:
```ruby
payload.key?('iss') && payload['iss'] == Setting['oidc_issuer'])
```
(i.e. just return the output of the condition instead of return false unless it's true)